network/src/storage.rs

```diff
@@ -0,0 +1,164 @@
+use crate::primitives::{Storage, StorageError};
+use async_trait::async_trait;
+use fs2::FileExt;
+use futures::stream::*;
+use tokio::{fs, io::AsyncWriteExt, task::spawn_blocking};
+
+pub struct StorageImpl {
+    lock_file: Option<std::fs::File>,
+    path: String,
+}
+
+#[async_trait]
+impl Storage for StorageImpl {
+    async fn create(storage_directory: &str) -> Result<(), StorageError> {
+        let _ = fs::remove_dir_all(storage_directory).await;
+        fs::create_dir(storage_directory).await?;
+        fs::File::create(format!("{}/lock", storage_directory)).await?;
+        Ok(())
+    }
+
+    async fn open(storage_directory: &str) -> Result<Self, StorageError>
+    where
+        Self: Sized,
+    {
+        let storage_directory_ = storage_directory.to_owned();
+        let file =
+            spawn_blocking(move || std::fs::File::open(format!("{}/lock", storage_directory_)))
```

Comment:
Why do you use `std::fs` instead of tokio's?

Reply:
`fs2` library is for file locking, but it does not support tokio. That's why I used `spawn_blocking()`.

Reply:
ok